### PR TITLE
Adding RawImage component

### PR DESCRIPTION
**Pull request by aghassemi**
RawImage component which takes in raw bytes in the form of a Uint8List and decodes and renders
the bytes as an image.

### CODE REVIEW

File: sky/packages/sky/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:typed_data';
 import 'dart:ui' as ui;
 
 import 'package:flutter/rendering.dart';
```
Comment (krisgiesing) [inline sky/packages/sky/lib/src/widgets/basic.dart]: Re-export Uint8List since it's used in the interface?

Comment (aghassemi) [inline sky/packages/sky/lib/src/widgets/basic.dart]: Done

File: sky/packages/sky/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -1095,6 +1097,39 @@ class DefaultAssetBundle extends InheritedWidget {
...
+class RawImage extends StatelessComponent {
+  RawImage(
+      {Key key,
+      this.bytes,
+      this.width,
+      this.height,
+      this.colorFilter,
+      this.fit,
+      this.repeat: ImageRepeat.noRepeat,
+      this.centerSlice})
```
Comment (abarth) [inline sky/packages/sky/lib/src/widgets/basic.dart]: The style guide says to format the whitespace here a bit differently.  See `AssetImage` for the style we're using.

**Comment by abarth** [comment]
LGTM w/ style fix.

File: sky/packages/sky/lib/src/widgets/basic.dart (hunk 2 above)
Comment (aghassemi) [inline sky/packages/sky/lib/src/widgets/basic.dart]: Done

**Comment by abarth** [comment]
LGTM

**Pull request merged**